specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1740 @@
+      "post": {
+        "tags": [
+          "BenefitUtilizationSummariesAsync"
+        ],
+        "operationId": "SavingsPlanScope_GenerateBenefitUtilizationSummariesReportAsync",
+        "description": "Triggers generation of a benefit utilization summaries report for the provided savings plan.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "GenerateUtilizationSummariesReportBySavingsPlan": {
+            "$ref": "./examples/BenefitUtilizationSummaries/Async/GenerateBenefitUtilizationSummariesReportBySavingsPlan.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/savingsPlanOrderIdParameter"
+          },
+          {
+            "$ref": "#/parameters/savingsPlanIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "benefitUtilizationSummariesRequest",
+            "description": "Async Benefit Utilization Summary report to be created.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BenefitUtilizationSummariesRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The status of the generation of the benefit utilization summaries report. This could be in progress, failed, or completed.",
+            "schema": {
+              "$ref": "#/definitions/BenefitUtilizationSummariesOperationStatus"
+            }
+          },
+          "202": {
+            "description": "Accepted. Request will be processed. Use the location header to check the status.",
+            "headers": {
+              "Location": {
+                "description": "The URL to check the status of the asynchronous operation.",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "The amount of delay to use while the status of the operation is checked. The value is expressed in seconds.",
+                "format": "int32",
+                "type": "integer"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/benefitUtilizationSummariesOperationResults/{operationId}": {
```

Comment:
>benefitUtilizationSummariesOperationResults/{operationId} [](http://example.com/codeflow?start=104&length=57)

these do not need to be modeled in swagger.  Generated SDKs, etc... will follow the async operation headers by convention

Reply:
its also important you don't model them in this way.  The results of these operations are sensitive as they contain SAS URIs and fall under the purview of these new capabilities to keep them secure:  [https://eng.ms/docs/products/arm/api_contracts/asyncoperationsigningandvalidation](https://eng.ms/docs/products/arm/api_contracts/asyncoperationsigningandvalidation)

Reply:
We authorize the call to the operation results API as well as confirm the operation id was tied to the scope requested (billing account, billing profile, reservation order, savings plan order). Other async API in this RP include the operation results API. We would like to be consistent.